.github/workflows/build.yml

```
@@ -5,237 +5,62 @@ name: Build
 on: [push, pull_request, workflow_dispatch]
 
 jobs:
-
-  # https://github.com/MiyooCFW/uboot/blob/master/.github/workflows/main.yml
-  uboot:
-    runs-on: ubuntu-20.04
-    container:
-      image: miyoocfw/toolchain:master
+  build-image:
+    uses: MiyooCFW/buildroot/.github/workflows/build.yml@master
+    with:
+      submodule: buildroot      
+  docker:
+    needs: build-image
+    permissions:
+      contents: read
+      packages: write
+      # This is used to complete the identity challenge
+      # with sigstore/fulcio when running outside of PRs.
+      id-token: write
+    runs-on: ubuntu-22.04
+    env:
+      # github.repository as <account>/<repo>
+      IMAGE_NAME: ${{ github.repository }}
+      DOCKERHUB_USERNAME: miyoocfw
     steps:
     - uses: actions/checkout@v3
-    - run: |
-        git submodule update --init --recursive -- uboot
-        apt update && apt install -y libpython3-dev python3-distutils bison flex
-        cd uboot
-        make miyoo_defconfig 
-        make
-    - uses: actions/upload-artifact@v3
-      with:
-        name: uboot
-        path: uboot/u-boot-sunxi-with-spl.bin
-        if-no-files-found: error # 'error', 'warn', 'ignore'; defaults to `warn`
 
-  # https://github.com/MiyooCFW/sdcard/blob/master/.github/workflows/build.yml
-  boot-scr:
-    runs-on: ubuntu-20.04
-    container:
-      image: miyoocfw/toolchain:master
-    steps:
-    - uses: actions/checkout@v3
-    - run: |
-        git submodule update --init --recursive -- sdcard
-        cd sdcard
-        ./gen_boot_scr.sh
-    - uses: actions/upload-artifact@v3
-      with:
-        name: boot-scr
-        path: sdcard/boot # todo: only upload the compiled .scr files instead of the entire boot folder
-        if-no-files-found: error # 'error', 'warn', 'ignore'; defaults to `warn`
+    # download and extract the artefact
+    - uses: actions/download-artifact@v3
+      with: 
+        name: SDK (uClibc)
+    - run: tar -xvf arm-miyoo-linux-uclibcgnueabi_sdk-buildroot.tar.gz
 
-  # todo: 1-bit sd card variant
-  kernel:
-    uses: MiyooCFW/kernel/.github/workflows/build.yml@master
-    with:
-      submodule: kernel
-        
-  # https://github.com/MiyooCFW/logo/blob/master/.github/workflows/c-cpp.yml
-  logo:
-    runs-on: ubuntu-20.04
-    container:
-      image: miyoocfw/toolchain:master
-    steps:
-    - uses: actions/checkout@v3
-    - run: |
-        git submodule update --init --recursive -- logo
-        cd logo
-        make -f Makefile.miyoo
-    - uses: actions/upload-artifact@v3
+    - name: Log into docker hub
+      #if: github.event_name != 'pull_request'
+      uses: docker/login-action@v1
```

Oh, it looks like we should upgrade this to v2 to handle warnings

```suggestion
      uses: docker/login-action@v2
```